home_work_6/index.js

```diff
@@ -0,0 +1,134 @@
+const checkIfValueValid = (variable) => {
+  if (variable == null || isNaN(variable)) {
+    return false;
+  }
+  return true;
+};
+
+//1.Написати функцію, яка приймає 4 аргументи і шукає найменше число серед них
+
+const showMin = (num1, num2, num3, num4) => {
+  const numArray = [].concat(num1, num2, num3, num4);
+  const filteredArr = numArray.filter((el) => checkIfValueValid(el));
+  if (filteredArr.length !== 0) {
+    return `${Math.min(...filteredArr)} найменше число з вказаних`;
+  } else {
+    return "Введіть числа для обробки";
+  }
+};
+console.log(showMin());
+
+/*
+2.Написати функцію, яка приймає 2 аргументи та шукає найбільший спільний дільник для них.
+Якщо такого числа немає, то повідомляти про відсутність
+*/
+
+const getGreatestCommonDivider = (num1, num2) => {
+  const arrDividersNum1 = [];
+
+  if (!checkIfValueValid(num1) || !checkIfValueValid(num2)) {
```

Comment:
Also you should validate numbers to 0 or validate `arrCommonDividers` length because if one of the arguments will be 0 `arrCommonDividers` will be empty and you will get `undefined` as a greatest common divider